cohortextractor2/dsl.py

```diff
@@ -221,10 +213,51 @@ def __hash__(self) -> int:
         return hash(repr(self.value))
 
 
+class Predicate:
+    def __init__(
+        self, column: Column, operator: str, other: str | bool | Codelist | None
+    ) -> None:
+        self._column = column
+        self._operator = operator
+        self._other = other
+
+    def apply_to(self, table: BaseTable) -> BaseTable:
+        return table.filter(self._column.name, **{self._operator: self._other})
+
+
 @dataclass
 class Column:
     name: str
 
+    def is_not_null(self):
+        return Predicate(self, "not_equals", None)
+
+
+class IdColumn(Column):
+    ...
+
+
+class BoolColumn(Column):
+    def is_true(self) -> Predicate:
+        return Predicate(self, "equals", True)
+
+
+class DateColumn(Column):
+    def __gt__(self, other: str) -> Predicate:
+        return Predicate(self, "greater_than", other)
+
+    def __lt__(self, other: str) -> Predicate:
+        return Predicate(self, "less_than", other)
```

Comment:
Should this have other operators too?  It's reasonable to expect to be able do <=, >=, ==, != on dates.  Possibly it'd make sense for there to be a base Column for shared operator methods - I'd expect `DateColumn` and `IntColumn` at least to share some of these methods.

Reply:
Also a date-specific `betweeen` method?

Reply:
Oh yes, absolutely. I just added enough to satisfy the existing tests. You're right that I should chuck in a bunch more while I'm at it.

Reply:
> Possibly it'd make sense for there to be a base Column for shared operator methods

Yes, I wondered about that. Or about having mixins for groups of operators. Or something. I'll try filling out the set of operators and then see if it makes sense to refactor.